app/src/main/java/com/sdp/movemeet/HomeScreenActivity.java

```diff
@@ -1,54 +1,41 @@
 package com.sdp.movemeet;
 
-import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.content.Intent;
 import android.os.Bundle;
-import android.util.Log;
 import android.view.View;
 
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
 
 import com.sdp.movemeet.map.GPSRecordingActivity;
 
-import com.google.android.gms.tasks.OnFailureListener;
-import com.google.android.gms.tasks.OnSuccessListener;
-import com.google.firebase.firestore.DocumentSnapshot;
-import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.Query;
-import com.google.firebase.firestore.QuerySnapshot;
-import com.sdp.movemeet.Activity.Activity;
-import com.sdp.movemeet.Backend.BackendActivityManager;
-
-import com.sdp.movemeet.map.MapsActivity;
 import com.sdp.movemeet.utility.ActivitiesUpdater;
 
-import java.util.ArrayList;
-import java.util.Date;
-
-import static com.sdp.movemeet.Backend.BackendActivityManager.ACTIVITIES_COLLECTION;
-import static com.sdp.movemeet.Sport.Running;
-
 public class HomeScreenActivity extends AppCompatActivity {
 
+    private FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home_screen);
+
         ActivitiesUpdater updater = ActivitiesUpdater.getInstance();
         updater.updateListActivities();
     }
 
     public void signIn(View v) {
-        startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // redirecting the user to the "Login" activity
+        //if (user != null) startActivity(new Intent(this, MainActivity.class));
```

Comment:
We can remove this commented line in a future commit.

Reply:
I left it here because I felt we had to add something similar at some point, i.e if the user is already logged in, there is no need to bring him to the LoginActivity. It would just be cumbersome for the user. We should directly bring him to the main activity. I commented it because it made the tests fail, but I think it would be important to think about it at some point.